server/src/main/java/org/elasticsearch/search/sort/SortAndFormats.java

```diff
@@ -11,17 +11,12 @@
 import org.apache.lucene.search.Sort;
 import org.elasticsearch.search.DocValueFormat;
 
-public final class SortAndFormats {
+public record SortAndFormats(Sort sort, DocValueFormat[] formats) {
```

Comment:
this and the other change are still causing noise in this PR. Could you make these a record in a specific PR where nothing needs review as it's only a mechanical change?